File: leetcode/python/0219_contains_duplicate_II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        index = {}
        for i in range(len(nums)):
            if nums[i] in index and abs(i - index[nums[i]]) <= k: return True
            index[nums[i]] = i
        return False

File: leetcode/cpp/0219_contains_duplicate_II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> occurred;

        for (int i = 0; i < nums.size(); i++) {
            if (occurred.find(nums[i]) != occurred.end()) {
                if (abs(i - occurred.find(nums[i])->second) <= k) {
                    return true;
                }
                else {
                    auto index = occurred.find(nums[i]);
                    index->second = i;
                }
            }
            else if (occurred.find(nums[i]) == occurred.end()) {
                occurred.insert(make_pair(nums[i], i));
            }
        }

        return false;
    }
};